apps/test/unit/code-studio/components/libraries/LibraryPublisherTest.js

```diff
@@ -0,0 +1,286 @@
+      librarySource: librarySource
+    };
+    onPublishSuccess = sinon.stub();
+    onUnpublishSuccess = sinon.stub();
+  });
+
+  afterEach(() => {
+    onPublishSuccess.resetHistory();
+    onUnpublishSuccess.resetHistory();
+  });
+
+  describe('On Load', () => {
+    it('suggests a default library name', () => {
+      libraryDetails.libraryName = 'foo';
+      let wrapper = shallow(
+        <LibraryPublisher
+          onPublishSuccess={onPublishSuccess}
+          onUnpublishSuccess={onUnpublishSuccess}
+          libraryDetails={libraryDetails}
+          clientApi={clientApi}
+        />
+      );
+
+      expect(wrapper.state().libraryName).to.equal(libraryName);
+    });
+
+    it('disables checkbox for items without comments', () => {
+      libraryDetails.sourceFunctionList.push({
+        functionName: 'bar',
+        comment: ''
+      });
+      let wrapper = shallow(
+        <LibraryPublisher
+          onPublishSuccess={onPublishSuccess}
+          onUnpublishSuccess={onUnpublishSuccess}
+          libraryDetails={libraryDetails}
+          clientApi={clientApi}
+        />
+      );
+
+      let checkboxes = wrapper.find(CHECKBOX_SELECTOR);
+      expect(checkboxes.first().prop('disabled')).to.be.false;
+      expect(checkboxes.last().prop('disabled')).to.be.true;
+    });
+
+    it('checks checkboxes of selected functions', () => {
+      libraryDetails.sourceFunctionList.push({
+        functionName: 'bar',
+        comment: 'comment'
+      });
+      libraryDetails.sourceFunctionList.push({
+        functionName: 'baz',
+        comment: ''
+      });
+      libraryDetails.selectedFunctions = {bar: true, baz: true};
+      let wrapper = shallow(
+        <LibraryPublisher
+          onPublishSuccess={onPublishSuccess}
+          onUnpublishSuccess={onUnpublishSuccess}
+          libraryDetails={libraryDetails}
+          clientApi={clientApi}
+        />
+      );
+
+      let checkboxes = wrapper.find(CHECKBOX_SELECTOR);
+      expect(checkboxes.at(0).prop('disabled')).to.be.false;
+      expect(checkboxes.at(1).prop('disabled')).to.be.false;
+      expect(checkboxes.at(2).prop('disabled')).to.be.true;
+      expect(checkboxes.at(0).prop('checked')).to.be.false;
+      expect(checkboxes.at(1).prop('checked')).to.be.true;
+      expect(checkboxes.at(2).prop('checked')).to.be.false;
+    });
+  });
+
+  describe('publish', () => {
+    let wrapper;
+    const description = 'description';
+    const selectedFunctions = {foo: true};
+    beforeEach(() => {
+      wrapper = shallow(
+        <LibraryPublisher
+          onPublishSuccess={onPublishSuccess}
+          onUnpublishSuccess={onUnpublishSuccess}
+          libraryDetails={libraryDetails}
+          clientApi={clientApi}
+        />
+      );
+    });
+
+    it('is disabled when no functions are checked', () => {
+      wrapper.setState({
+        libraryDescription: description
+      });
+
+      wrapper.instance().publish();
```

Comment:
similar to my comment above ([link](https://github.com/code-dot-org/code-dot-org/pull/32716/files#r368135523)), i think we should interact with what's rendered by the component here by finding the publish button and simulating a click on it

Reply:
Enzyme generally discourages using `simulate` and instead asks developers to directly call the function props (full write-up here if you are interested https://github.com/airbnb/enzyme/issues/1357). They have even said that they will be deprecating `simulate` in v4. I do agree that we should simulate clicking the publish button and verifying what happens, but that should be in a UI test rather than a unit test.

Reply:
oh wow, TIL! that's really good to know. this [comment](https://github.com/airbnb/enzyme/issues/1357#issuecomment-403999415) resonates with me: 

> a proper simulate would be most useful - but since one doesn't exist, the current broken one should be avoided

forget what i said 😄